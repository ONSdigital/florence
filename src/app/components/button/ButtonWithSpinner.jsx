import React from "react";

import PropTypes from "prop-types";

const propTypes = {
    isSubmitting: PropTypes.bool.isRequired,
    buttonText: PropTypes.string.isRequired
};

const ButtonWithSpinner = props => (
    <div>
        <button type="submit" className="btn btn--primary margin-top--1" disabled={props.isSubmitting}>
            {props.isSubmitting ? <div className="loader loader--dark" /> : props.buttonText}
        </button>
    </div>
);
ButtonWithSpinner.propTypes = propTypes;
export default ButtonWithSpinner;
