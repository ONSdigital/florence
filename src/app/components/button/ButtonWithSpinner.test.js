import React from "react";
import ButtonWithSpinner from "./ButtonWithSpinner";
import renderer from "react-test-renderer";
import {mount} from "enzyme";

test("Button to be enabled, and spinner to not be present when not submitting", () => {
    const props = {
        isSubmitting: false
    };
    let component = mount(<ButtonWithSpinner {...props}/>);
    let btn = component.find(".btn")
    expect(btn.length).toBe(1);
    expect(btn.props().disabled).toBe(false);
    expect(component.find(".form__loader").length).toBe(0);

    component = renderer.create(<ButtonWithSpinner {...props}/>);
    expect(component.toJSON()).toMatchSnapshot();
});

test("Button to be enabled, and spinner to not be present when submitting", () => {
    const props = {
        isSubmitting: true
    };
    let component = mount(<ButtonWithSpinner {...props}/>);
    let btn = component.find(".btn")
    expect(btn.length).toBe(1);
    expect(btn.props().disabled).toBe(true);
    expect(component.find(".form__loader").length).toBe(1);

    component = renderer.create(<ButtonWithSpinner {...props}/>);
    expect(component.toJSON()).toMatchSnapshot();
});
