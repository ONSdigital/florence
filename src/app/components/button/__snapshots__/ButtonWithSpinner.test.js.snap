// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to be enabled, and spinner to be present when submitting 1`] = `
<div>
  <button
    className="btn btn--primary margin-top--1"
    disabled={true}
    type="submit"
  >
    <div
      className="loader loader--dark"
    />
  </button>
</div>
`;

exports[`Button to be enabled, and spinner to not be present when not submitting 1`] = `
<div>
  <button
    className="btn btn--primary margin-top--1"
    disabled={false}
    type="submit"
  >
    foo
  </button>
</div>
`;
